sdk/storage/azure-storage-file-share/tests/test_share.py

```diff
@@ -1654,6 +1656,118 @@ def test_share_paid_bursting(self, **kwargs):
+
+        first_share_client.set_share_properties(access_tier='Hot')
+        first_share_props = first_share_client.get_share_properties()
+        second_share_client.set_share_properties(access_tier='Cool')
+        second_share_props = second_share_client.get_share_properties()
+        assert first_share_props is not None
+        assert first_share_props.name == first_share.share_name
+        assert first_share_props.access_tier == 'Hot'
+        assert second_share_props is not None
+        assert second_share_props.name == second_share.share_name
+        assert second_share_props.access_tier == 'Cool'
+
+        first_share_deleted = first_share_client.delete_share()
+        second_share_deleted = second_share_client.delete_share()
+        assert first_share_deleted is None
+        assert second_share_deleted is None
+        with pytest.raises(ResourceNotFoundError):
+            first_share_client.get_share_properties()
+        with pytest.raises(ResourceNotFoundError):
+            second_share_client.get_share_properties()
+
+    @FileSharePreparer()
+    @recorded_by_proxy
+    def test_share_lease_with_oauth(self, **kwargs):
+        storage_account_name = kwargs.pop("storage_account_name")
+        storage_account_key = kwargs.pop("storage_account_key")
+        token_credential = self.get_credential(ShareClient)
+
+        # Arrange
+        self._setup(storage_account_name, storage_account_key)
+        share = self._create_share('test')
+        share_client = ShareClient(
+            self.account_url(storage_account_name, "file"),
+            share_name=share.share_name,
+            credential=token_credential,
+            token_intent=TEST_INTENT
+        )
+
+        # Act / Assert
+        lease_duration = 15
+        lease = share_client.acquire_lease(
+            lease_id='00000000-1111-2222-3333-444444444444',
+            lease_duration=lease_duration
+        )
+        props = share_client.get_share_properties(lease=lease)
+        assert props.lease.duration == 'fixed'
+        assert props.lease.state == 'leased'
+        assert props.lease.status == 'locked'
+
+        lease_id1 = lease.id
+        new_lease_id = '29e0b239-ecda-4f69-bfa3-95f6af91464c'
+        lease.change(proposed_lease_id=new_lease_id)
+        lease_id2 = lease.id
+        assert lease_id1 is not None
+        assert lease_id2 is not None
+        assert lease_id1 != lease_id2
+        assert lease_id2 == new_lease_id
+
+        lease.renew()
+        assert lease.id == new_lease_id
+
+        time_elapsed = 2
+        self.sleep(time_elapsed)
+        with pytest.raises(HttpResponseError):
+            share_client.delete_share()
+        lease_break_period = 14
+        time_to_acquire_lease = lease.break_lease(lease_break_period=lease_break_period)
+        assert time_to_acquire_lease < min(lease_duration - time_elapsed, lease_break_period)
+
+        result = lease.release()
+        assert result is None
```

Comment:
Let's simplify this test a bit by removing all the stuff about `lease.change` and `lease.break`. Just have acquire, renew, and release. Also can remove the assert this comment is on (line 1765) because it doesn't really check anything of value.